src/main/java/dk/cngroup/trainings/spring/springassignment/controller/dto/user/UserSignupDTO.java

```diff
@@ -0,0 +1,74 @@
+package dk.cngroup.trainings.spring.springassignment.controller.dto.user;
+
+import javax.persistence.Column;
+import javax.validation.constraints.*;
+
+public class UserSignupDTO {
+
+	@NotNull
+	@Size(min = 2, max = 20, message = "first name must be between 2 and 20 characters")
+	private String firstName;
+	@NotNull
+	@Size(min = 2, max = 20, message = "last name must be between 2 and 20 characters")
+	private String lastName;
+	@NotNull
+	@Column(unique = true)
+	@Email
+	@Pattern(regexp = ".+@.+\\.[a-z]+", message = "invalid email address")
+	private String email;
+	@NotNull
+	@Size(min = 4, message = "password must be at least 4 characters")
+	private String password;
+	private String matchingPassword;
```

Comment:
you can add constraint for matching passwords too (see Skype). But AFAIK it will have to be on the class level, not on field level.

Reply:
Yeah, I will look for way to implement it.
Class or field, if it gets the job done, then I take it.